src/Confluent.Kafka/Consumer.cs

```diff
@@ -133,30 +133,30 @@ private void LogCallback(IntPtr rk, SyslogLevel level, string fac, string buf)
 
             if (err == ErrorCode.Local_AssignPartitions)
             {
-                if (partitionsAssignedHandler != null)
+                if (rebalanceHandler != null)
                 {
                     assignCallCount = 0;
-                    partitionsAssignedHandler(partitionList);
+                    rebalanceHandler(new RebalanceEvent(partitionAssignment, true));
                     if (assignCallCount == 1) { return; }
```

Comment:
add a comment saying why we do this

Reply:
there's a comment above the if block that covers this (which probably didn't make it into the bounds of the pr diff).